### DISCUSSION TITLE
Specific mesh not being a manifold

### DISCUSSION THREAD

**Discussion by deltakosh** [elalish/manifold#1011] [Q&A] [answered]
Hey team!!

After the introduction of Manifold for Babylon.js, we are getting report of some GLTF not being "a manifold".

How can we detect that and fix it?

Here is one for instance: https://dl.dropbox.com/scl/fi/0z9plcnggrfmnjk9fh87h/S1.glb?rlkey=ifplgssotvoiuyge4cob29iuo

The playground where you can see the error happening:
https://playground.babylonjs.com/#M829XG#32

Please note that it works with very similar (but obviously different meshes):
https://playground.babylonjs.com/#M829XG#32

**Comment by elalish** [comment, marked as answer]
Indeed, most glTFs, in fact most meshes in general are not manifold (solid / watertight). Detection is easy, fixing is hard (in general). Manifold provides the `Merge` function that will handle simple cases that at least "look" manifold. I even made a glTF extension to keep manifoldness from getting lost in the format - it has some background and explanation: https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Vendor/EXT_mesh_manifold/README.md

I also have a drag-and-drop page that uses our merge function on glTFs, so you can see what it can fix and what it gives up on: https://manifoldcad.org/make-manifold

There are lots of manifold fixing programs around, especially for 3D printing, but they're all full of heuristics because it can be very hard to know what solid someone is hoping to represent with an open mesh.

**Comment by deltakosh** [comment]
Ok I see thanks a lot for the explanation (I was already using merge so this is on the DCC tools side now to get a better mesh)
